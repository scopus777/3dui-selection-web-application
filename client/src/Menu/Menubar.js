import React, {Component} from 'react';
import {LinkContainer} from "react-router-bootstrap";
import {Nav, Navbar} from "react-bootstrap";

/*
* This Component is Always shown and generates the Menubar in the top of the screen
* */
class Menubar extends Component {

    render() {
        return (
            <Navbar bg="light" expand="lg">
                <Navbar.Brand href="/">S3DIT</Navbar.Brand>
                <Navbar.Toggle aria-controls="basic-navbar-nav"/>
                <Navbar.Collapse id="basic-navbar-nav">
                    <Nav className="mr-auto">
                        {/*LinkContainer have to be used to make the link work with React-Router*/}
                        <LinkContainer exact to="/" activename="active">
                            <Nav.Link>Filter</Nav.Link>
                        </LinkContainer>
                        <LinkContainer exact to="/sort/" activename="active">
                            <Nav.Link>Sort</Nav.Link>
                        </LinkContainer>
                        <LinkContainer exact to="/help/" activename="active">
                            <Nav.Link>Help</Nav.Link>
                        </LinkContainer>
                    </Nav>
                    <Nav pullright="true">
                        <LinkContainer exact to="/submit/" activename="active">
                            <Nav.Link className="justify-content-end">Submit Technique</Nav.Link>
                        </LinkContainer>
                    </Nav>
                </Navbar.Collapse>
            </Navbar>
        );
    }
}

export default Menubar;
