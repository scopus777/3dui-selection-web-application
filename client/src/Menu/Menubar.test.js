import React from 'react';
import Menubar from './Menubar';
import {create} from 'react-test-renderer'
import {MemoryRouter} from "react-router-dom";

describe('Test Menubar Snapshot', () => {
    test('testing menubar snapshot', () => {
        let tree = create(
            <MemoryRouter>
                <Menubar active="1"/>
            </MemoryRouter>);
        expect(tree.toJSON()).toMatchSnapshot();
    })
});
