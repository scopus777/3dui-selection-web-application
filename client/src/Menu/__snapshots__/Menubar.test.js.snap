// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Menubar Snapshot testing menubar snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <a
    className="navbar-brand"
    href="/"
  >
    VR-Recommender
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        activename="active"
        className="active nav-link"
        data-rb-event-key="/"
        href="/"
        onClick={[Function]}
        onKeyDown={[Function]}
        style={Object {}}
      >
        Filter
      </a>
      <a
        activename="active"
        className="nav-link"
        data-rb-event-key="/help/"
        href="/help/"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Help
      </a>
    </div>
  </div>
</nav>
`;
